File: Cpp/1678_Goal_Parser_Interpretation.cpp
class Solution {
public:
    string interpret(string command) {
        string result = "";
        
        for(int i = 0; i < command.size(); ++i) {
            if(command[i] == 'G') {
                result += 'G';
            }else if(command[i] == ')') {
                if(command[i - 1] == '(') result += 'o';
                else result += "al";
            }
        }
        
        return result;
    }
};

File: Kotlin/1678_Goal_Parser_Interpretation.kt
class Solution {
    fun interpret(command: String): String {
        var result = StringBuilder()
        
        for(i in command.indices) {
            if(command[i] == 'G') {
                result.append('G')
            }else if(command[i] == ')') {
                if(command[i - 1] == '(') result.append('o')
                else result.append("al")
            }
        }
        
        return result.toString()
    }
}